src/watchmaker/utils/imds/detect/__init__.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+"""Detect  module."""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals, with_statement)
+
+import concurrent.futures
+import logging
+
+from watchmaker.utils.imds.detect.providers.aws_provider import AWSProvider
+from watchmaker.utils.imds.detect.providers.azure_provider import AzureProvider
+from watchmaker.utils.imds.detect.providers.provider import AbstractProvider
+
+log = logging.getLogger(__name__)
+
+MAX_WORKERS = 10
+
+
+def provider():
+    """Identify and return identifier."""
+    exception_list = []
+    results = []
+    futures = []
+    with concurrent.futures.ThreadPoolExecutor(
+            max_workers=MAX_WORKERS) as executor:
+        futures.append(executor.submit(identify, AWSProvider))
+        futures.append(executor.submit(identify, AzureProvider))
+
+    concurrent.futures.wait(futures)
+    for fut in futures:
+        try:
+            result = fut.result()
+            if result != AbstractProvider.identifier:
```

Comment:
Under what condition(s) would `AbstractProvider.identifier` be returned, given that we're specifcally calling only for AWSProvider and AzureProvider?

Reply:
Adding a comment in the code, but a non-cloud environment or unsupported cloud env, for example gcp. The cloud detect module would not detect AWS or AZURE and it would return "unknown".  We are attempting to detect if we are on AWS or AZURE as they are supported, but it does not mean we are on them.

Reply:
I guess I was just expecting them to return nothing in that case, or an exception. I was thinking of it as something like "assert that I am a valid provider".

Reply:
Maybe something like:

```
        try:
            results.append(fut.result())
        except InvalidProvider:
            pass
        except BaseException as ex:
            exception_list.append(str(ex))
```